### PR TITLE
feat: other players loading progress: replacing owner-net-variables with server rpcs

### PR DESCRIPTION
**Pull request by LPLafontaineB**
### Description

This draft PR shows an alternative solution to MTT-2239, using server-authoritative network variables instead of owner-writable ones. We can decide if we prefer this solution over the other one (#580 )
### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] JIRA ticket ID is in the PR title or at least one commit message
 - [ ] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/LoadingProgressManager.cs [hunk 1]
```diff
@@ -11,42 +11,56 @@ namespace Unity.Multiplayer.Samples.Utilities
...
             {
-                if (IsSpawned && ProgressTrackers.ContainsKey(NetworkManager.LocalClientId))
+                if (IsSpawned)
                 {
-                    ProgressTrackers[NetworkManager.LocalClientId].Progress.Value = value;
-                }
-                else
-                {
-                    m_LocalProgress = value;
+                    UpdateClientProgressServerRpc(NetworkManager.LocalClientId, value);
```
Comment (SamuelBellomo) [inline Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/LoadingProgressManager.cs:53]: should extract this, shouldn't put heavy operations in properties

Comment (SamuelBellomo) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/LoadingProgressManager.cs:53]: need to keep track if value changed or not

Comment (SamuelBellomo) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/LoadingProgressManager.cs:53]: need to keep track if value changed or not

**Comment by LPLafontaineB** [comment]
Closing this because we decided the first solution was better

**Pull request closed without merging**